pandas/core/internals/blocks.py

```diff
@@ -2114,10 +2143,26 @@ def get_values(self, dtype: DtypeObj | None = None) -> np.ndarray:
     def values_for_json(self) -> np.ndarray:
         return self.values
 
+    @cache_readonly
```

Comment:
mypy doesn't like these usages of `cache_readonly`:

```
pandas/core/internals/blocks.py:2147: error: Cannot override writeable attribute with read-only property  [override]
pandas/core/internals/blocks.py:2154: error: Cannot override writeable attribute with read-only property  [override]
```

I could make these use `property` but that seems silly. I'm also confused why this isn't an issue for `ExtensionBlock`, which also has an `is_numeric` implementation decorated with `cache_readonly` and mypy is fine with that. Maybe a mypy bug?

Reply:
its fine to do a "type: ignore" on these